packages/artifact/src/internal/path-and-artifact-name-validation.ts

```diff
@@ -0,0 +1,82 @@
+import {info} from '@actions/core'
+
+/**
+ * Invalid characters that cannot be in the artifact name or an uploaded file. Will be rejected
+ * from the server if attempted to be sent over. These characters are not allowed due to limitations with certain
+ * file systems such as NTFS. To maintain platform-agnostic behavior, all characters that are not supported by an
+ * individual filesystem/platform will not be supported on all fileSystems/platforms
+ *
+ * FilePaths can include characters such as \ and / which are not permitted in the artifact name alone
+ */
+const invalidArtifactFilePathCharacters = new Map<string, string>([
+  ['"', 'Double quote "'],
+  [':', ' Colon :'],
+  ['<', ' Less than <'],
+  ['>', ' Greater than >'],
+  ['|', ' Vertical bar |'],
+  ['*', ' Asterisk *'],
+  ['?', ' Question mark ?'],
+  ['\r', ' Carriage return \\r'],
+  ['\n', ' Line feed \\n']
```

Comment:
nit: most of these messages have spaces before them except Double quote?

Reply:
That's intentional. It was to make the display nicer since calling `.toString()` adds a comma between entry

This is what it looked like before the extra spaces, I initially played with adding them at the end:
![image](https://user-images.githubusercontent.com/16109154/144302903-94f300a4-a9af-492f-bd4b-8c0c7b35cbea.png)

The first one doesn't have one so the text in the annotations lines up on the left side

I've added a comment for clarification

Reply:
I've removed the extra space. For context I had to add spaces because calling `.toString()` made the output a bit ugly with commas right next to each other. Double quote was the first thing in the output 

![image](https://user-images.githubusercontent.com/16109154/144304991-f941d854-8b04-47c1-8271-ef0aa8241752.png)


What it looks like now.

![image](https://user-images.githubusercontent.com/16109154/144304800-93c10790-0b98-4444-971a-9a58241020b3.png)

Reply:
we prob could've done `.join(", ")` instead of the `.toString()` on the array